src/fieldofview.rs

```diff
@@ -24,7 +24,7 @@ pub fn field_of_view(start: Point, range: i32, fov_check: &dyn Algorithm2D) -> V
         scan_fov_line(start, Point::new(right, y), range_squared, fov_check, &mut visible_points);
     }
 
-    let mut result = Vec::new();
+    let mut result = Vec::with_capacity(visible_points.len());
```

Comment:
I think this optimization is not necessary if you use `collect`.

Reply:
Can't collect since the HashMap contains pointers to Points, not Points. Maybe if we do `.map(|p| *p).collect()` ?

Reply:
It works when using `into_iter`, see #28.

Reply:
ohh neat, thanks!